huxley/api/mixins.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2011-2015 Berkeley Model United Nations. All rights reserved.
+# Use of this source code is governed by a BSD License (see LICENSE).
+
+import json
+
+from django.db import transaction
+from django.http import QueryDict
+
+from huxley.core.models import Delegate
+
+from rest_framework import status
+from rest_framework.response import Response
+
+class ListUpdateModelMixin(object):
+    """
+    Update a queryset
+    """
+
+    def list_update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', False)
+        data = request.data
+        if isinstance(data, QueryDict):
+            data = json.loads(request.data.items()[0][0])
+
+        response_data = []
+        delegate_ids = []
+        updates = {}
+
+        for obj in data:
```

Comment:
Let's go simpler with this. Assuming the data is an array of delegates, we can convert it to a dictionary with a dictionary comprehension:
```python
updates = {delegate['id']: delegate for delegate in data}
```
Then, in the transaction, we can use the keys of the dictionary:
```python
delegates = Delegate.objects.filter(id__in=updates.keys())
```